Sources/Profile-View.php

```diff
@@ -690,7 +693,7 @@ function showAlerts($memID)
 	}
 
 	// Prepare the pagination vars.
-	$maxIndex = 10;
+	$maxIndex = !empty($modSettings['alerts_per_page']) && (int) $modSettings['alerts_per_page'] < 1000 ? min((int) $modSettings['alerts_per_page'], 1000) : 25;
```

Comment:
$modSettings needs to be added to the global variables declaration in this function.